DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/Operations/Send/RootObjectBuilder.cs

```diff
@@ -24,29 +26,26 @@ public RootObjectBuilder(IUnitOfWorkFactory unitOfWorkFactory)
   }
 
   public Base Build(
-    ISendFilter sendFilter,
+    IEnumerable<RhinoObject> objects,
+    SendInfo sendInfo,
     Action<string, double?>? onOperationProgressed = null,
     CancellationToken ct = default
   )
   {
-    List<RhinoObject> rhinoObjects = sendFilter
-      .GetObjectIds()
-      .Select(id => RhinoDoc.ActiveDoc.Objects.FindId(new Guid(id)))
-      .Where(obj => obj != null)
-      .ToList();
-
-    if (rhinoObjects.Count == 0)
+    IEnumerable<RhinoObject> rhinoObjects = objects as RhinoObject[] ?? objects.ToArray();
```

Comment:
I'm confused too now..

Looked into it, and it was [CA1851](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1851). Will have a look again

Reply:
Imo, if we're going to keep calling .ToList or .ToArray. we should have just typed things as an `IReadonlyList` or `IList` in the first place.
The `Build` and `ConvertObjects` function should take an `IReadonlyList` rather than an `IEnumerable`. 
and things suddenly become cleaner...